In [4]:
import pandas as pd
import numpy as np

Import file csv,excel,etc.

In [5]:
df = pd.read_csv('https://raw.githubusercontent.com/ardhiraka/PFDS_sources/master/nbaallelo.csv')

Melihat jumlah data

In [9]:
len(df)

126314

Melihat jumlah data dan kolom

In [8]:
df.shape

(126314, 23)

Tampilkan 5 data teratas dan terbawah

In [10]:
df.head()

gameorder       game_id lg_id  _iscopy  year_id  date_game  seasongame  \
0          1  194611010TRH   NBA        0     1947  11/1/1946           1   
1          1  194611010TRH   NBA        1     1947  11/1/1946           1   
2          2  194611020CHS   NBA        0     1947  11/2/1946           1   
3          2  194611020CHS   NBA        1     1947  11/2/1946           2   
4          3  194611020DTF   NBA        0     1947  11/2/1946           1   

   is_playoffs team_id  fran_id  ...  win_equiv  opp_id  opp_fran  opp_pts  \
0            0     TRH  Huskies  ...  40.294830     NYK    Knicks       68   
1            0     NYK   Knicks  ...  41.705170     TRH   Huskies       66   
2            0     CHS    Stags  ...  42.012257     NYK    Knicks       47   
3            0     NYK   Knicks  ...  40.692783     CHS     Stags       63   
4            0     DTF  Falcons  ...  38.864048     WSC  Capitols       50   

   opp_elo_i  opp_elo_n  game_location  game_result  forecast notes  
0  1300.0000  1306.7233              H            L  0.640065   NaN  
1  1300.0000  1293.2767              A            W  0.359935   NaN  
2  1306.7233  1297.0712              H            W  0.631101   NaN  
3  1300.0000  1309.6521              A            L  0.368899   NaN  
4  1300.0000  1320.3811              H            L  0.640065   NaN  

[5 rows x 23 columns]

In [11]:
df.tail()

gameorder       game_id lg_id  _iscopy  year_id  date_game  \
126309      63155  201506110CLE   NBA        0     2015  6/11/2015   
126310      63156  201506140GSW   NBA        0     2015  6/14/2015   
126311      63156  201506140GSW   NBA        1     2015  6/14/2015   
126312      63157  201506170CLE   NBA        0     2015  6/16/2015   
126313      63157  201506170CLE   NBA        1     2015  6/16/2015   

        seasongame  is_playoffs team_id    fran_id  ...  win_equiv  opp_id  \
126309         100            1     CLE  Cavaliers  ...  60.309792     GSW   
126310         102            1     GSW   Warriors  ...  68.013329     CLE   
126311         101            1     CLE  Cavaliers  ...  60.010067     GSW   
126312         102            1     CLE  Cavaliers  ...  59.290245     GSW   
126313         103            1     GSW   Warriors  ...  68.519516     CLE   

         opp_fran  opp_pts  opp_elo_i  opp_elo_n  game_location  game_result  \
126309   Warriors      103  1790.9591  1809.9791              H            L   
126310  Cavaliers       91  1704.3949  1700.7391              H            W   
126311   Warriors      104  1809.9791  1813.6349              A            L   
126312   Warriors      105  1813.6349  1822.2881              H            L   
126313  Cavaliers       97  1700.7391  1692.0859              A            W   

        forecast notes  
126309  0.546572   NaN  
126310  0.765565   NaN  
126311  0.234435   NaN  
126312  0.481450   NaN  
126313  0.518550   NaN  

[5 rows x 23 columns]

Cek info dari file csv

In [12]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 126314 entries, 0 to 126313
Data columns (total 23 columns):
 #   Column         Non-Null Count   Dtype  
---  ------         --------------   -----  
 0   gameorder      126314 non-null  int64  
 1   game_id        126314 non-null  object 
 2   lg_id          126314 non-null  object 
 3   _iscopy        126314 non-null  int64  
 4   year_id        126314 non-null  int64  
 5   date_game      126314 non-null  object 
 6   seasongame     126314 non-null  int64  
 7   is_playoffs    126314 non-null  int64  
 8   team_id        126314 non-null  object 
 9   fran_id        126314 non-null  object 
 10  pts            126314 non-null  int64  
 11  elo_i          126314 non-null  float64
 12  elo_n          126314 non-null  float64
 13  win_equiv      126314 non-null  float64
 14  opp_id         126314 non-null  object 
 15  opp_fran       126314 non-null  object 
 16  opp_pts        126314 non-null  int64  
 17  opp_elo_i      126314 non-nul

Melihat nilai yang adap pada file csv

In [13]:
df.describe()

gameorder        _iscopy        year_id     seasongame  \
count  126314.000000  126314.000000  126314.000000  126314.000000   
mean    31579.000000       0.500000    1988.200374      43.533733   
std     18231.927643       0.500002      17.582309      25.375178   
min         1.000000       0.000000    1947.000000       1.000000   
25%     15790.000000       0.000000    1975.000000      22.000000   
50%     31579.000000       0.500000    1990.000000      43.000000   
75%     47368.000000       1.000000    2003.000000      65.000000   
max     63157.000000       1.000000    2015.000000     108.000000   

         is_playoffs            pts          elo_i          elo_n  \
count  126314.000000  126314.000000  126314.000000  126314.000000   
mean        0.063857     102.729982    1495.236055    1495.236055   
std         0.244499      14.814845     112.139945     112.461687   
min         0.000000       0.000000    1091.644500    1085.774400   
25%         0.000000      93.000000    1417.237975    1416.994900   
50%         0.000000     103.000000    1500.945550    1500.954400   
75%         0.000000     112.000000    1576.060000    1576.291625   
max         1.000000     186.000000    1853.104500    1853.104500   

           win_equiv        opp_pts      opp_elo_i      opp_elo_n  \
count  126314.000000  126314.000000  126314.000000  126314.000000   
mean       41.707889     102.729982    1495.236055    1495.236055   
std        10.627332      14.814845     112.139945     112.461687   
min        10.152501       0.000000    1091.644500    1085.774400   
25%        34.103035      93.000000    1417.237975    1416.994900   
50%        42.113357     103.000000    1500.945550    1500.954400   
75%        49.635328     112.000000    1576.060000    1576.291625   
max        71.112038     186.000000    1853.104500    1853.104500   

            forecast  
count  126314.000000  
mean        0.500000  
std         0.215252  
min         0.020447  
25%         0.327989  
50%         0.500000  
75%         0.672011  
max         0.979553

EDA (Exploration Data Analyst)

In [14]:
#cek jumlah tiap tim
df['team_id'].value_counts()

BOS    5997
NYK    5769
LAL    5078
DET    4985
PHI    4533
       ... 
INJ      60
PIT      60
DTF      60
TRH      60
SDS      11
Name: team_id, Length: 104, dtype: int64

In [15]:
#cek jumlah data kolom fran_id
df["fran_id"].value_counts()

Lakers          6024
Celtics         5997
Knicks          5769
Warriors        5657
Pistons         5650
Sixers          5644
Hawks           5572
Kings           5475
Wizards         4582
Spurs           4309
Bulls           4307
Pacers          4227
Thunder         4178
Rockets         4154
Nuggets         4120
Nets            4106
Suns            4080
Bucks           4034
Trailblazers    3870
Cavaliers       3810
Clippers        3733
Jazz            3555
Mavericks       3013
Heat            2371
Pelicans        2254
Magic           2207
Timberwolves    2131
Grizzlies       1657
Raptors         1634
Hornets          894
Colonels         846
Squires          799
Spirits          777
Stars            756
Sounds           697
Baltimore        467
Floridians       440
Condors          430
Capitols         291
Olympians        282
Sails            274
Stags            260
Bombers          249
Steamrollers     168
Packers           72
Redskins          65
Rebels            63
Waterloo     

tim bernama "Lakers" memainkan 6024 pertandingan, tetapi hanya 5078 yang dimainkan oleh Los Angeles Lakers. Cari tahu siapa tim "Lakers" lainnya:

In [16]:
df.loc[df["fran_id"] == "Lakers", "team_id"].value_counts()

LAL    5078
MNL     946
Name: team_id, dtype: int64

Kapan tim MNL memainkan pertaningan pertama dan terakhir

In [17]:
df.loc[df["team_id"] == "MNL", "date_game"].agg(("min", "max"))

min    1/1/1949
max    4/9/1959
Name: date_game, dtype: object

Cek jumlah point yang didapat oleh tim BOS selama mamainkan pertandingan basket

In [18]:
df.loc[df["team_id"] == "BOS", "pts"].sum()

626484

Understanding Series Objects

In [19]:
revenues = pd.Series([5555,7000,1980])
revenues

0    5555
1    7000
2    1980
dtype: int64

In [20]:
revenues.values

array([5555, 7000, 1980])

create series

In [21]:
city_revenues = pd.Series(
    [4200, 8000, 6500],
    index=["Amsterdam", "Toronto", "Tokyo"]
)
city_revenues

Amsterdam    4200
Toronto      8000
Tokyo        6500
dtype: int64

In [22]:
city_employee_count = pd.Series({"Amsterdam": 5, "Tokyo": 8})
city_employee_count

Amsterdam    5
Tokyo        8
dtype: int64

In [23]:
city_employee_count.keys()

Index(['Amsterdam', 'Tokyo'], dtype='object')

In [24]:
"Tokyo" in city_employee_count

True

In [25]:
city_data = pd.DataFrame({
    "revenue": city_revenues,
    "employee_count": city_employee_count
})
city_data

revenue  employee_count
Amsterdam     4200             5.0
Tokyo         6500             8.0
Toronto       8000             NaN

In [26]:
city_data.index

Index(['Amsterdam', 'Tokyo', 'Toronto'], dtype='object')

In [27]:
city_data.values

array([[4.2e+03, 5.0e+00],
       [6.5e+03, 8.0e+00],
       [8.0e+03,     nan]])

In [28]:
city_data.axes

[Index(['Amsterdam', 'Tokyo', 'Toronto'], dtype='object'),
 Index(['revenue', 'employee_count'], dtype='object')]

In [29]:
city_data.axes[0]

Index(['Amsterdam', 'Tokyo', 'Toronto'], dtype='object')

In [30]:
city_data.keys()

Index(['revenue', 'employee_count'], dtype='object')

In [32]:
df.head()

gameorder       game_id lg_id  _iscopy  year_id  date_game  seasongame  \
0          1  194611010TRH   NBA        0     1947  11/1/1946           1   
1          1  194611010TRH   NBA        1     1947  11/1/1946           1   
2          2  194611020CHS   NBA        0     1947  11/2/1946           1   
3          2  194611020CHS   NBA        1     1947  11/2/1946           2   
4          3  194611020DTF   NBA        0     1947  11/2/1946           1   

   is_playoffs team_id  fran_id  ...  win_equiv  opp_id  opp_fran  opp_pts  \
0            0     TRH  Huskies  ...  40.294830     NYK    Knicks       68   
1            0     NYK   Knicks  ...  41.705170     TRH   Huskies       66   
2            0     CHS    Stags  ...  42.012257     NYK    Knicks       47   
3            0     NYK   Knicks  ...  40.692783     CHS     Stags       63   
4            0     DTF  Falcons  ...  38.864048     WSC  Capitols       50   

   opp_elo_i  opp_elo_n  game_location  game_result  forecast notes  
0  1300.0000  1306.7233              H            L  0.640065   NaN  
1  1300.0000  1293.2767              A            W  0.359935   NaN  
2  1306.7233  1297.0712              H            W  0.631101   NaN  
3  1300.0000  1309.6521              A            L  0.368899   NaN  
4  1300.0000  1320.3811              H            L  0.640065   NaN  

[5 rows x 23 columns]

In [33]:
df.index

RangeIndex(start=0, stop=126314, step=1)

In [34]:
df.axes

[RangeIndex(start=0, stop=126314, step=1),
 Index(['gameorder', 'game_id', 'lg_id', '_iscopy', 'year_id', 'date_game',
        'seasongame', 'is_playoffs', 'team_id', 'fran_id', 'pts', 'elo_i',
        'elo_n', 'win_equiv', 'opp_id', 'opp_fran', 'opp_pts', 'opp_elo_i',
        'opp_elo_n', 'game_location', 'game_result', 'forecast', 'notes'],
       dtype='object')]

In [35]:
city_revenues

Amsterdam    4200
Toronto      8000
Tokyo        6500
dtype: int64

In [36]:
city_revenues["Toronto"]

8000

In [37]:
city_revenues[1]

8000

In [39]:
colors = pd.Series(
    ["red", "purple", "blue", "green", "yellow"],
    index=[1, 2, 3, 5, 8]
)
colors

1       red
2    purple
3      blue
5     green
8    yellow
dtype: object

In [40]:
colors.loc[1]

'red'

In [42]:
city_data.loc["Amsterdam"]

revenue           4200.0
employee_count       5.0
Name: Amsterdam, dtype: float64

In [43]:
city_data.loc["Tokyo": "Toronto"]

revenue  employee_count
Tokyo       6500             8.0
Toronto     8000             NaN

In [44]:
city_data.iloc[1]

revenue           6500.0
employee_count       8.0
Name: Tokyo, dtype: float64

In [45]:
df.iloc[-2]

gameorder               63157
game_id          201506170CLE
lg_id                     NBA
_iscopy                     0
year_id                  2015
date_game           6/16/2015
seasongame                102
is_playoffs                 1
team_id                   CLE
fran_id             Cavaliers
pts                        97
elo_i               1700.7391
elo_n               1692.0859
win_equiv           59.290245
opp_id                    GSW
opp_fran             Warriors
opp_pts                   105
opp_elo_i           1813.6349
opp_elo_n           1822.2881
game_location               H
game_result                 L
forecast              0.48145
notes                     NaN
Name: 126312, dtype: object

In [46]:
city_data.loc["Amsterdam": "Tokyo", "revenue"]

Amsterdam    4200
Tokyo        6500
Name: revenue, dtype: int64

In [47]:
df

gameorder       game_id lg_id  _iscopy  year_id  date_game  \
0               1  194611010TRH   NBA        0     1947  11/1/1946   
1               1  194611010TRH   NBA        1     1947  11/1/1946   
2               2  194611020CHS   NBA        0     1947  11/2/1946   
3               2  194611020CHS   NBA        1     1947  11/2/1946   
4               3  194611020DTF   NBA        0     1947  11/2/1946   
...           ...           ...   ...      ...      ...        ...   
126309      63155  201506110CLE   NBA        0     2015  6/11/2015   
126310      63156  201506140GSW   NBA        0     2015  6/14/2015   
126311      63156  201506140GSW   NBA        1     2015  6/14/2015   
126312      63157  201506170CLE   NBA        0     2015  6/16/2015   
126313      63157  201506170CLE   NBA        1     2015  6/16/2015   

        seasongame  is_playoffs team_id    fran_id  ...  win_equiv  opp_id  \
0                1            0     TRH    Huskies  ...  40.294830     NYK   
1                1            0     NYK     Knicks  ...  41.705170     TRH   
2                1            0     CHS      Stags  ...  42.012257     NYK   
3                2            0     NYK     Knicks  ...  40.692783     CHS   
4                1            0     DTF    Falcons  ...  38.864048     WSC   
...            ...          ...     ...        ...  ...        ...     ...   
126309         100            1     CLE  Cavaliers  ...  60.309792     GSW   
126310         102            1     GSW   Warriors  ...  68.013329     CLE   
126311         101            1     CLE  Cavaliers  ...  60.010067     GSW   
126312         102            1     CLE  Cavaliers  ...  59.290245     GSW   
126313         103            1     GSW   Warriors  ...  68.519516     CLE   

         opp_fran  opp_pts  opp_elo_i  opp_elo_n  game_location  game_result  \
0          Knicks       68  1300.0000  1306.7233              H            L   
1         Huskies       66  1300.0000  1293.2767              A            W   
2          Knicks       47  1306.7233  1297.0712              H            W   
3           Stags       63  1300.0000  1309.6521              A            L   
4        Capitols       50  1300.0000  1320.3811              H            L   
...           ...      ...        ...        ...            ...          ...   
126309   Warriors      103  1790.9591  1809.9791              H            L   
126310  Cavaliers       91  1704.3949  1700.7391              H            W   
126311   Warriors      104  1809.9791  1813.6349              A            L   
126312   Warriors      105  1813.6349  1822.2881              H            L   
126313  Cavaliers       97  1700.7391  1692.0859              A            W   

        forecast notes  
0       0.640065   NaN  
1       0.359935   NaN  
2       0.631101   NaN  
3       0.368899   NaN  
4       0.640065   NaN  
...          ...   ...  
126309  0.546572   NaN  
126310  0.765565   NaN  
126311  0.234435   NaN  
126312  0.481450   NaN  
126313  0.518550   NaN  

[126314 rows x 23 columns]

In [48]:
current_decade = df[df["year_id"] > 2010]
current_decade

gameorder       game_id lg_id  _iscopy  year_id   date_game  \
113656      56829  201010260BOS   NBA        1     2011  10/26/2010   
113657      56829  201010260BOS   NBA        0     2011  10/26/2010   
113658      56830  201010260LAL   NBA        1     2011  10/26/2010   
113659      56830  201010260LAL   NBA        0     2011  10/26/2010   
113660      56831  201010260POR   NBA        1     2011  10/26/2010   
...           ...           ...   ...      ...      ...         ...   
126309      63155  201506110CLE   NBA        0     2015   6/11/2015   
126310      63156  201506140GSW   NBA        0     2015   6/14/2015   
126311      63156  201506140GSW   NBA        1     2015   6/14/2015   
126312      63157  201506170CLE   NBA        0     2015   6/16/2015   
126313      63157  201506170CLE   NBA        1     2015   6/16/2015   

        seasongame  is_playoffs team_id    fran_id  ...  win_equiv  opp_id  \
113656           1            0     MIA       Heat  ...  45.141479     BOS   
113657           1            0     BOS    Celtics  ...  53.754559     MIA   
113658           1            0     HOU    Rockets  ...  40.900764     LAL   
113659           1            0     LAL     Lakers  ...  55.608730     HOU   
113660           1            0     PHO       Suns  ...  53.879230     POR   
...            ...          ...     ...        ...  ...        ...     ...   
126309         100            1     CLE  Cavaliers  ...  60.309792     GSW   
126310         102            1     GSW   Warriors  ...  68.013329     CLE   
126311         101            1     CLE  Cavaliers  ...  60.010067     GSW   
126312         102            1     CLE  Cavaliers  ...  59.290245     GSW   
126313         103            1     GSW   Warriors  ...  68.519516     CLE   

            opp_fran  opp_pts  opp_elo_i  opp_elo_n  game_location  \
113656       Celtics       88  1625.0994  1629.3031              A   
113657          Heat       80  1547.3638  1543.1600              H   
113658        Lakers      112  1647.6016  1649.2002              A   
113659       Rockets      110  1504.1956  1502.5969              H   
113660  Trailblazers      106  1552.8141  1565.2162              A   
...              ...      ...        ...        ...            ...   
126309      Warriors      103  1790.9591  1809.9791              H   
126310     Cavaliers       91  1704.3949  1700.7391              H   
126311      Warriors      104  1809.9791  1813.6349              A   
126312      Warriors      105  1813.6349  1822.2881              H   
126313     Cavaliers       97  1700.7391  1692.0859              A   

        game_result  forecast notes  
113656            L  0.264418   NaN  
113657            W  0.735582   NaN  
113658            L  0.197632   NaN  
113659            W  0.802368   NaN  
113660            L  0.485912   NaN  
...             ...       ...   ...  
126309            L  0.546572   NaN  
126310            W  0.765565   NaN  
126311            L  0.234435   NaN  
126312            L  0.481450   NaN  
126313            W  0.518550   NaN  

[12658 rows x 23 columns]

In [49]:
current_decade.shape

(12658, 23)

In [50]:
df[
    (df["_iscopy"] == 0) &
    (df["team_id"].str.startswith("LA")) &
    (df["year_id"] == 1992) &
    (df["notes"].notnull())
    ]

gameorder       game_id lg_id  _iscopy  year_id date_game  seasongame  \
68901      34451  199205030LAC   NBA        0     1992  5/3/1992          86   
68903      34452  199205030LAL   NBA        0     1992  5/3/1992          86   

       is_playoffs team_id   fran_id  ...  win_equiv  opp_id      opp_fran  \
68901            1     LAC  Clippers  ...  47.888252     UTA          Jazz   
68903            1     LAL    Lakers  ...  39.395412     POR  Trailblazers   

       opp_pts  opp_elo_i  opp_elo_n  game_location  game_result  forecast  \
68901      107  1647.5092  1638.4095              H            W  0.499125   
68903      102  1655.1210  1670.3677              H            L  0.406874   

                                            notes  
68901  at Anaheim, CA (Anaheim Convention Center)  
68903                            at Las Vegas, NV  

[2 rows x 23 columns]

In [51]:
df.head(3)

gameorder       game_id lg_id  _iscopy  year_id  date_game  seasongame  \
0          1  194611010TRH   NBA        0     1947  11/1/1946           1   
1          1  194611010TRH   NBA        1     1947  11/1/1946           1   
2          2  194611020CHS   NBA        0     1947  11/2/1946           1   

   is_playoffs team_id  fran_id  ...  win_equiv  opp_id  opp_fran  opp_pts  \
0            0     TRH  Huskies  ...  40.294830     NYK    Knicks       68   
1            0     NYK   Knicks  ...  41.705170     TRH   Huskies       66   
2            0     CHS    Stags  ...  42.012257     NYK    Knicks       47   

   opp_elo_i  opp_elo_n  game_location  game_result  forecast notes  
0  1300.0000  1306.7233              H            L  0.640065   NaN  
1  1300.0000  1293.2767              A            W  0.359935   NaN  
2  1306.7233  1297.0712              H            W  0.631101   NaN  

[3 rows x 23 columns]

In [52]:
df.groupby("fran_id", sort=False)["pts"].sum()

fran_id
Huskies           3995
Knicks          582497
Stags            20398
Falcons           3797
Capitols         22387
Celtics         626484
Steamrollers     12372
Ironmen           3674
Bombers          17793
Rebels            4474
Warriors        591224
Baltimore        37219
Jets              4482
Pistons         572758
Lakers          637444
Kings           569245
Hawks           567261
Denver            4818
Olympians        22864
Redskins          5372
Waterloo          4921
Packers           6193
Sixers          585891
Wizards         474809
Bulls           437269
Thunder         437735
Squires          91127
Stars            84940
Rockets         432504
Colonels         94435
Pacers          438288
Nuggets         445780
Spurs           453822
Spirits          85874
Sounds           75582
Floridians       49568
Nets            417809
Condors          49642
Bucks           418326
Suns            437486
Clippers        380523
Cavaliers       380416
Trailblazers    402695
Sai

In [53]:
df[
    (df["fran_id"] == "Spurs") &
    (df["year_id"] > 2010)
    ].groupby(["year_id", "game_result"])["game_id"].count()

year_id  game_result
2011     L              25
         W              63
2012     L              20
         W              60
2013     L              30
         W              73
2014     L              27
         W              78
2015     L              31
         W              58
Name: game_id, dtype: int64

In [54]:
df[
    (df["fran_id"] == "Warriors") &
    (df["year_id"] == 2015)
    ].groupby(["is_playoffs", "game_result"])["game_id"].count()

is_playoffs  game_result
0            L              15
             W              67
1            L               5
             W              16
Name: game_id, dtype: int64

In [55]:
df

gameorder       game_id lg_id  _iscopy  year_id  date_game  \
0               1  194611010TRH   NBA        0     1947  11/1/1946   
1               1  194611010TRH   NBA        1     1947  11/1/1946   
2               2  194611020CHS   NBA        0     1947  11/2/1946   
3               2  194611020CHS   NBA        1     1947  11/2/1946   
4               3  194611020DTF   NBA        0     1947  11/2/1946   
...           ...           ...   ...      ...      ...        ...   
126309      63155  201506110CLE   NBA        0     2015  6/11/2015   
126310      63156  201506140GSW   NBA        0     2015  6/14/2015   
126311      63156  201506140GSW   NBA        1     2015  6/14/2015   
126312      63157  201506170CLE   NBA        0     2015  6/16/2015   
126313      63157  201506170CLE   NBA        1     2015  6/16/2015   

        seasongame  is_playoffs team_id    fran_id  ...  win_equiv  opp_id  \
0                1            0     TRH    Huskies  ...  40.294830     NYK   
1                1            0     NYK     Knicks  ...  41.705170     TRH   
2                1            0     CHS      Stags  ...  42.012257     NYK   
3                2            0     NYK     Knicks  ...  40.692783     CHS   
4                1            0     DTF    Falcons  ...  38.864048     WSC   
...            ...          ...     ...        ...  ...        ...     ...   
126309         100            1     CLE  Cavaliers  ...  60.309792     GSW   
126310         102            1     GSW   Warriors  ...  68.013329     CLE   
126311         101            1     CLE  Cavaliers  ...  60.010067     GSW   
126312         102            1     CLE  Cavaliers  ...  59.290245     GSW   
126313         103            1     GSW   Warriors  ...  68.519516     CLE   

         opp_fran  opp_pts  opp_elo_i  opp_elo_n  game_location  game_result  \
0          Knicks       68  1300.0000  1306.7233              H            L   
1         Huskies       66  1300.0000  1293.2767              A            W   
2          Knicks       47  1306.7233  1297.0712              H            W   
3           Stags       63  1300.0000  1309.6521              A            L   
4        Capitols       50  1300.0000  1320.3811              H            L   
...           ...      ...        ...        ...            ...          ...   
126309   Warriors      103  1790.9591  1809.9791              H            L   
126310  Cavaliers       91  1704.3949  1700.7391              H            W   
126311   Warriors      104  1809.9791  1813.6349              A            L   
126312   Warriors      105  1813.6349  1822.2881              H            L   
126313  Cavaliers       97  1700.7391  1692.0859              A            W   

        forecast notes  
0       0.640065   NaN  
1       0.359935   NaN  
2       0.631101   NaN  
3       0.368899   NaN  
4       0.640065   NaN  
...          ...   ...  
126309  0.546572   NaN  
126310  0.765565   NaN  
126311  0.234435   NaN  
126312  0.481450   NaN  
126313  0.518550   NaN  

[126314 rows x 23 columns]

In [56]:
nba = df.copy()

In [57]:
nba.shape

(126314, 23)

In [58]:
nba["difference"] = df.pts - df.opp_pts
nba

gameorder       game_id lg_id  _iscopy  year_id  date_game  \
0               1  194611010TRH   NBA        0     1947  11/1/1946   
1               1  194611010TRH   NBA        1     1947  11/1/1946   
2               2  194611020CHS   NBA        0     1947  11/2/1946   
3               2  194611020CHS   NBA        1     1947  11/2/1946   
4               3  194611020DTF   NBA        0     1947  11/2/1946   
...           ...           ...   ...      ...      ...        ...   
126309      63155  201506110CLE   NBA        0     2015  6/11/2015   
126310      63156  201506140GSW   NBA        0     2015  6/14/2015   
126311      63156  201506140GSW   NBA        1     2015  6/14/2015   
126312      63157  201506170CLE   NBA        0     2015  6/16/2015   
126313      63157  201506170CLE   NBA        1     2015  6/16/2015   

        seasongame  is_playoffs team_id    fran_id  ...  opp_id   opp_fran  \
0                1            0     TRH    Huskies  ...     NYK     Knicks   
1                1            0     NYK     Knicks  ...     TRH    Huskies   
2                1            0     CHS      Stags  ...     NYK     Knicks   
3                2            0     NYK     Knicks  ...     CHS      Stags   
4                1            0     DTF    Falcons  ...     WSC   Capitols   
...            ...          ...     ...        ...  ...     ...        ...   
126309         100            1     CLE  Cavaliers  ...     GSW   Warriors   
126310         102            1     GSW   Warriors  ...     CLE  Cavaliers   
126311         101            1     CLE  Cavaliers  ...     GSW   Warriors   
126312         102            1     CLE  Cavaliers  ...     GSW   Warriors   
126313         103            1     GSW   Warriors  ...     CLE  Cavaliers   

        opp_pts  opp_elo_i  opp_elo_n game_location  game_result  forecast  \
0            68  1300.0000  1306.7233             H            L  0.640065   
1            66  1300.0000  1293.2767             A            W  0.359935   
2            47  1306.7233  1297.0712             H            W  0.631101   
3            63  1300.0000  1309.6521             A            L  0.368899   
4            50  1300.0000  1320.3811             H            L  0.640065   
...         ...        ...        ...           ...          ...       ...   
126309      103  1790.9591  1809.9791             H            L  0.546572   
126310       91  1704.3949  1700.7391             H            W  0.765565   
126311      104  1809.9791  1813.6349             A            L  0.234435   
126312      105  1813.6349  1822.2881             H            L  0.481450   
126313       97  1700.7391  1692.0859             A            W  0.518550   

        notes difference  
0         NaN         -2  
1         NaN          2  
2         NaN         16  
3         NaN        -16  
4         NaN        -17  
...       ...        ...  
126309    NaN        -21  
126310    NaN         13  
126311    NaN        -13  
126312    NaN         -8  
126313    NaN          8  

[126314 rows x 24 columns]

In [59]:
nba.shape

(126314, 24)

In [60]:
nba["difference"].max()

68

In [61]:
renamed_nba = nba.rename(
    columns={"game_result": "result", "game_location": "location"}
)

renamed_nba.head()

gameorder       game_id lg_id  _iscopy  year_id  date_game  seasongame  \
0          1  194611010TRH   NBA        0     1947  11/1/1946           1   
1          1  194611010TRH   NBA        1     1947  11/1/1946           1   
2          2  194611020CHS   NBA        0     1947  11/2/1946           1   
3          2  194611020CHS   NBA        1     1947  11/2/1946           2   
4          3  194611020DTF   NBA        0     1947  11/2/1946           1   

   is_playoffs team_id  fran_id  ...  opp_id  opp_fran  opp_pts  opp_elo_i  \
0            0     TRH  Huskies  ...     NYK    Knicks       68  1300.0000   
1            0     NYK   Knicks  ...     TRH   Huskies       66  1300.0000   
2            0     CHS    Stags  ...     NYK    Knicks       47  1306.7233   
3            0     NYK   Knicks  ...     CHS     Stags       63  1300.0000   
4            0     DTF  Falcons  ...     WSC  Capitols       50  1300.0000   

   opp_elo_n location  result  forecast  notes difference  
0  1306.7233        H       L  0.640065    NaN         -2  
1  1293.2767        A       W  0.359935    NaN          2  
2  1297.0712        H       W  0.631101    NaN         16  
3  1309.6521        A       L  0.368899    NaN        -16  
4  1320.3811        H       L  0.640065    NaN        -17  

[5 rows x 24 columns]